content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+
+```python
+class MyMEDMnistTrial(PyTorchTrial):
+    def __init__(self, context: PyTorchTrialContext) -> None:
+        self.context = context
+
+        self.info = INFO[self.context.get_hparam("data_flag")]
+        task = self.info["task"]
+        n_classes = len(self.info["label"])
+
+        self.context = context
+        if self.context.get_hparam("model_flag") == "resnet18":
+            model = resnet18(pretrained=False, num_classes=n_classes)
+        elif self.context.get_hparam("model_flag") == "resnet50":
+            model = resnet50(pretrained=False, num_classes=n_classes)
+        else:
+            raise NotImplementedError
+
+        self.model = self.context.wrap_model(model)
+
+        optimizer = torch.optim.Adam(
+            self.model.parameters(), lr=self.context.get_hparam("lr")
+        )
+        self.optimizer = self.context.wrap_optimizer(optimizer)
+
+        if self.context.get_hparam("task") == "multi-label, binary-class":
+            self.criterion = nn.BCEWithLogitsLoss()
+        else:
+            self.criterion = nn.CrossEntropyLoss()
+```
+
+#### Step 1.3 Port Data Loaders  
+
+Next, we’ll port the training and evaluation data loaders to the following class functions as follows. This is the code that trains one batch of data. You no longer need the standard for loop to iterate over batches or epochs inside these functions, since Determined will handle the training loop for you. 
+
+```python
+    def build_training_data_loader(self) -> DataLoader:
+        DataClass = getattr(medmnist, self.info["python_class"])
+
+        if self.context.get_hparam("resize"):
+            data_transform = transforms.Compose(
+                [
+                    transforms.Resize((224, 224), interpolation=PIL.Image.NEAREST),
+                    transforms.ToTensor(),
+                    transforms.Normalize(mean=[0.5], std=[0.5]),
+                ]
+            )
+        else:
+            data_transform = transforms.Compose(
+                [transforms.ToTensor(), transforms.Normalize(mean=[0.5], std=[0.5])]
+            )
+
+        train_dataset = DataClass(
+            split="train",
+            transform=data_transform,
+            download=False,
+            as_rgb=True,
+            root=DATASET_ROOT,
+        )
+        train_loader = determined.pytorch.DataLoader(
+            dataset=train_dataset,
+            batch_size=self.context.get_per_slot_batch_size(),
+            shuffle=True,
+        )
+
+        return train_loader
+```
+
+In each function, we initialize and return the relevant PyTorch DataLoader object (e.g. `val_loader from build_validation_data_loader`, and `train_loader in build_training_data_loader`).  
+
+#### Step 1.4 Port Training and Evaluation Functions 
```

Comment:
Header should be:
"Step 1.4: Port training and evaluation functions" [again, I also added the colon, but just to be consistent with your first two steps. It's up to you whether or not you want to keep the colons in these steps. Just make sure you are consistent with all of them.]